### PR TITLE
cpu/cortexm: cleanup cortexm.inc.mk

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR cleans up the cortexm.inc.mk file:
- it fixes the Makefile indentation to make it more readable
- it removes the unnecessary use of export
- there's also a small simplification when setting the MCPU variable.

### Testing procedure

- A green Murdock is ok
- Verify a firmware built for an ARM target is unchanged:

<details><summary>This PR:</summary>

```
$ RIOT_VERSION=test make BOARD=nucleo-l073rz -C examples/hello-world clean all
[...]
   text	   data	    bss	    dec	    hex	filename
   8356	    120	   2568	  11044	   2b24	/work/riot/RIOT/examples/hello-world/bin/nucleo-l073rz/hello-world.elf
```

</details>

<details><summary>master:</summary>

```
$ RIOT_VERSION=test make BOARD=nucleo-l073rz -C examples/hello-world clean all
[...]
   text	   data	    bss	    dec	    hex	filename
   8356	    120	   2568	  11044	   2b24	/work/riot/RIOT/examples/hello-world/bin/nucleo-l073rz/hello-world.elf
```

</details>

### Issues/PRs references

Related to #10850 (for LINKFLAGS)

### CODE REVIEW

File: makefiles/arch/cortexm.inc.mk [hunk 1]
```diff
@@ -26,22 +26,22 @@ CFLAGS_OPT  ?= -Os
...
+LINKFLAGS += -T$(LINKER_SCRIPT) -Wl,--fatal-warnings
 
-export LINKFLAGS += $(CFLAGS_CPU) $(CFLAGS_DBG) $(CFLAGS_OPT) -static -lgcc -nostartfiles
-export LINKFLAGS += -Wl,--gc-sections
+LINKFLAGS += $(CFLAGS_CPU) $(CFLAGS_DBG) $(CFLAGS_OPT) -static -lgcc -nostartfiles
+LINKFLAGS += -Wl,--gc-sections
 
 # Tell the build system that the CPU depends on the Cortex-M common files:
-export USEMODULE += cortexm_common
+USEMODULE += cortexm_common
```
Comment (fjmolinas) [inline makefiles/arch/cortexm.inc.mk]: USEMODULE is already exported in vars, so OK.

File: makefiles/arch/cortexm.inc.mk [hunk 2]
```diff
@@ -3,20 +3,20 @@ ifeq (,$(CPU_MODEL))
 endif
 
 # Target triple for the build. Use arm-none-eabi if you are unsure.
-export TARGET_ARCH ?= arm-none-eabi
+TARGET_ARCH ?= arm-none-eabi
```
Comment (fjmolinas) [inline makefiles/arch/cortexm.inc.mk]: Already exported in vars.in.mk so OK.

**Comment by fjmolinas** [comment]
Needs rebasing as well.

File: makefiles/arch/cortexm.inc.mk [hunk 3]
```diff
@@ -73,61 +73,62 @@ CFLAGS += -DCPU_ARCH_$(call uppercase_and_underscore,$(CPU_ARCH))
...
+else
+  CFLAGS_FPU ?= -mfloat-abi=soft
+endif
+
+ifeq ($(CPU_ARCH),cortex-m4f)
+  MCPU = cortex-m4
 else
-CFLAGS_FPU ?= -mfloat-abi=soft
-export MCPU ?= $(CPU_ARCH)
+  MCPU ?= $(CPU_ARCH)
```
[the remark this answers is not in the document]
Comment (aabadie) [reply makefiles/arch/cortexm.inc.mk]: Thanks for pointing this. It was the only thing I was unsure how to tackle it correctly. Using `target-export-variables` did the trick.

File: pkg/jerryscript/Makefile.jerryscript [hunk 4]
```diff
@@ -31,7 +31,7 @@ all: libjerry
 libjerry:
 	cmake -B$(JERRYSCRIPT_BUILD_DIR) -H./ \
 	 -DCMAKE_SYSTEM_NAME=RIOT \
-	 -DCMAKE_SYSTEM_PROCESSOR="$(MCPU)" \
+	 -DCMAKE_SYSTEM_PROCESSOR:FORCE="$(MCPU)" \
```
Comment (fjmolinas) [inline pkg/jerryscript/Makefile.jerryscript]: Why did you need to add FORCE?

Comment (aabadie) [reply pkg/jerryscript/Makefile.jerryscript]: Ah, I forgot to remove this... It's not needed, I'll remove it.

Comment (fjmolinas) [reply pkg/jerryscript/Makefile.jerryscript]: Ok, squash afterwards!

Comment (fjmolinas) [reply pkg/jerryscript/Makefile.jerryscript]: ups there was nothing to squash.

Comment (aabadie) [reply pkg/jerryscript/Makefile.jerryscript]: Wait, I still have to push the update ;)

Comment (aabadie) [reply pkg/jerryscript/Makefile.jerryscript]: Done

**Comment by fjmolinas** [comment]
My ACK, still stands.

**Comment by aabadie** [comment]
I don't think the [Murdock failure](https://ci.riot-os.org/RIOT-OS/RIOT/13017/3e3f4e232406f2ad74763fbf76943cad40cc9d41/output/run_test/examples/suit_update/nrf52dk:gnu.txt) during `CI: run tests` is related.

**Comment by fjmolinas** [comment]
> I don't think the Murdock failure during CI: run tests is related.

Ajj, the suit test is really flimsy lately. I'll try to find a solution for this.

**Comment by aabadie** [comment]
I'm rerunning Murdock without `CI: run tests`, there was only one failure in the previous build.
